build/build.sh

```diff
@@ -0,0 +1,36 @@
+#!/usr/bin/env bash
+set -e
+
+SBT_CMD="sbt ++${TRAVIS_SCALA_VERSION} validate"
+
+if [[ "${TRAVIS_PULL_REQUEST}" == "false" ]]; then
+    SBT_CMD+=" +coverageOff +publish"
+    openssl aes-256-cbc -pass env:ENCRYPTION_PASSWORD -in ./build/secring.gpg.enc -out local.secring.gpg -d
+    openssl aes-256-cbc -pass env:ENCRYPTION_PASSWORD -in ./build/pubring.gpg.enc -out local.pubring.gpg -d
+    openssl aes-256-cbc -pass env:ENCRYPTION_PASSWORD -in ./build/credentials.sbt.enc -out local.credentials.sbt -d
+    openssl aes-256-cbc -pass env:ENCRYPTION_PASSWORD -in ./build/deploy_key.pem.enc -out local.deploy_key.pem -d
+
+    if [[ "${TRAVIS_BRANCH}" == "master" && $(cat version.sbt) != *"SNAPSHOT"* ]]; then
```

Comment:
What does commit the new version back? Or maybe we aren't doing it yet?

Reply:
The magic seems to be part of the `sbt-release` functionality defined here
https://github.com/sbt/sbt-release/blob/v1.0.9/src/main/scala/ReleaseExtra.scala#L77-L93

Used as part of the release process defined here https://github.com/finagle/finch/pull/995/files#diff-fdc3abdfd754eeb24090dbd90aeec2ceR131

TBH, i haven't tested the release process yet.  I have an internal jfrog account where i can test with a sandbox repo. Maybe i can create a dummy repo to test out the versioning. I also need to read up a bit more on `sbt-release` and understand how the next version gets promoted.  In theory it should be `0.26.0-SNAPSHOT` when a new release is performed.  Maybe setting to `1.0.0-SNAPSHOT` is a manual step before committing to the `1.0.0` milestone

Wasn't super familiar with this plugin but seems like it can do a bit more than what is currently being done.  Seems like it can apparently automate release notes?